samples/samples/admin/backup_snippet.py

```diff
@@ -0,0 +1,556 @@
+    )
+
+
+# [END spanner_restore_backup_with_encryption_key]
+
+
+# [START spanner_cancel_backup_create]
+def cancel_backup(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+    expire_time = datetime.utcnow() + timedelta(days=30)
+
+    # Create a backup.
+    request = backup_pb.CreateBackupRequest(
+        parent=instance.name,
+        backup_id=backup_id,
+        backup=backup_pb.Backup(
+            database=database.name,
+            expire_time=expire_time,
+        ),
+    )
+
+    operation = spanner_client.database_admin_api.create_backup(request)
+    # Cancel backup creation.
+    operation.cancel()
+
+    # Cancel operations are best effort so either it will complete or
+    # be cancelled.
+    while not operation.done():
+        time.sleep(300)  # 5 mins
+
+    try:
+        spanner_client.database_admin_api.get_backup(
+            backup_pb.GetBackupRequest(
+                name="{}/backups/{}".format(instance.name, backup_id)
+            )
+        )
+    except NotFound:
+        print("Backup creation was successfully cancelled.")
+        return
+    print("Backup was created before the cancel completed.")
+    spanner_client.database_admin_api.delete_backup(
+        backup_pb.DeleteBackupRequest(
+            name="{}/backups/{}".format(instance.name, backup_id)
+        )
+    )
+    print("Backup deleted.")
+
+
+# [END spanner_cancel_backup_create]
+
+
+# [START spanner_list_backup_operations]
+def list_backup_operations(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the CreateBackup operations.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.CreateBackupMetadata) "
+        "AND (metadata.database:{})"
+    ).format(database_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
+            backup_pb.CreateBackupMetadata, op.metadata
+        )
+        print(
+            "Backup {} on database {}: {}% complete.".format(
+                metadata.name, metadata.database, metadata.progress.progress_percent
+            )
+        )
+
+    # List the CopyBackup operations.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/google.spanner.admin.database.v1.CopyBackupMetadata) "
+        "AND (metadata.source_backup:{})"
+    ).format(backup_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```

Comment:
same here

Reply:
All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:
Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_backup_operations.go#L64

JAVA= https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1683-L1689